jobs/mysql/templates/drain.sh

```diff
@@ -1,6 +1,59 @@
-#!/bin/bash -e
+#!/bin/bash -eu
 
-/var/vcap/packages/mariadb/support-files/mysql.server stop --pid-file=/var/vcap/sys/run/mysql/mysql.pid > /dev/null
-return_code=$?
-echo 0
-exit ${return_code}
+NODE_IP=<%= spec.ip %>
+MYSQL_PORT=<%= p("cf_mysql.mysql.port") %>
+
+LOG_DIR="/var/vcap/sys/log/mysql/"
+
+# if the node ain't running, ain't got nothin' to drain
+if ! ps -p $(</var/vcap/sys/run/mysql/mysql.pid) >/dev/null; then
+  echo "mysql is not running: drain OK" &>> "$LOG_DIR/drain.log"
+  echo 0; exit 0 # drain success
+fi
+
+function wsrep_var() {
+  local var_name=$1
+  local host=$2
+  local port=$3
+  if [[ $var_name =~ ^wsrep_[a-z_]+$ ]]; then
+    timeout 5 \
+      /usr/local/bin/mysql --defaults-file=/var/vcap/jobs/mysql/config/drain.cnf -h "$host" -P "$port" \
+      --execute="SHOW STATUS LIKE '$var_name'" -N |\
+      awk '{print $2}' | tr -d '\n'
+  fi
+}
+
+CLUSTER_NODES=(`wsrep_var wsrep_incoming_addresses $NODE_IP $MYSQL_PORT | sed -e 's/,/ /g'`)
```

Comment:
Please use Bash-isms consistenly, and use the `$()` syntax here instead of backticks.

Please quote `"$NODE_IP" "$MYSQL_PORT"` to be sure the exact values are passed.